File: 02_DevApplication_IOS_avecObjectif-C/07_MKMapItem/07_MKMapItem/ViewController.m
//
//  ViewController.m
//  07_MKMapItem
//
//  Created by etudiant on 20/04/2017.
//  Copyright © 2017 etudiant. All rights reserved.
//

#import "ViewController.h"
#import <CoreLocation/CoreLocation.h>
#import <MapKit/MapKit.h>
#import <Contacts/Contacts.h>

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *txtAdresse;
@property (weak, nonatomic) IBOutlet UITextField *txtVille;
@property (weak, nonatomic) IBOutlet UITextField *txtCp;
@property (weak, nonatomic) IBOutlet UITextField *txtPays;

@property (nonatomic)CLLocationCoordinate2D coordonnesDepart;
@property (nonatomic)CLLocationCoordinate2D coordonnesDestination;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}
- (IBAction)btnAfficherTrajetTouché:(id)sender {
    //récuprer une instance du service de géolocalisation
    CLGeocoder* geocoder = [[CLGeocoder alloc] init];
    //Créer une chaine contenant l'adresse de destination
    NSString* addr = [NSString stringWithFormat:@"%@ %@ %@ %@", self.txtAdresse.text, self.txtVille.text, self.txtCp.text, self.txtPays.text];
    
    NSLog(@"%@", addr);
    
    //passer l'adresse obtenue à partir de la saisie au service
    //de géolocalisation
    [geocoder geocodeAddressString:addr completionHandler:^(NSArray<CLPlacemark *> * _Nullable placemarks, NSError * _Nullable error) {
        //analyseer l'argument error pour savoir
        if(error)
        {
            NSLog(@"Echec de service geocode avec erreur %@", error);
            return;
        }
        //si geocoder a trouvé une adresse ou plusieurs
        //qui corresponde à la saisie données, il les met
        //dans un tableau qui nous est fourni dans l'argument
        //placemarks
        //si aucune adresse n'a été trouvée, placemaks contient nil
        if(placemarks && placemarks.count>0)
        {
            // choisir la premiere adresse trouvée
            CLPlacemark* placemark = placemarks[0];
            
            //récupérer les coordonnées géographiques de cette adresse
            
            float latitude = placemark.location.coordinate.latitude;
            float longitude = placemark.location.coordinate.longitude;
            
            NSLog(@"latitude: %f, longitude: %f", latitude, longitude);
            
            self.coordonnesDestination = placemark.location.coordinate;
            
            [self afficherMap];
        }
    }
     
     ];
}

-(void) afficherMap
{
    //créer un dictionnaire qui doit contenir les informations
    //saisie (c'est mes informations du point d'arrivée)
    //(du placemark) de notre trajet
    //ce dictionaire possède des clès spécifiques:
    NSDictionary* dicoDestination = @{
                                      CNPostalAddressCityKey:self.txtVille.text,
                                      CNPostalAddressStreetKey:self.txtAdresse.text,
                                      CNPostalAddressPostalCodeKey:self.txtCp.text,
                                      CNPostalAddressCountryKey:self.txtPays.text};
    
    MKPlacemark* placemarkDestination = [[MKPlacemark alloc] initWithCoordinate:self.coordonnesDestination addressDictionary:dicoDestination];
    MKMapItem* mkMapItem = [[MKMapItem alloc] initWithPlacemark:placemarkDestination];
    //preciser des eventuelles options de parcours
    //ces options doivent être données aussi dans un dictionnaire
    //avec des clés spécifiques
    NSDictionary* dicoOptions = @{MKLaunchOptionsDirectionsModeKey:MKLaunchOptionsDirectionsModeDriving};
    
    [mkMapItem openInMapsWithLaunchOptions:dicoOptions];
    
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 03_Swift_fondamentaux/08_MKMAPItem/08_MKMAPItem/ViewController.swift
//
//  ViewController.swift
//  08_MKMAPItem
//
//  Created by etudiant on 12/05/2017.
//  Copyright © 2017 etudiant. All rights reserved.
//

import UIKit
import CoreLocation
import Contacts
import MapKit

class ViewController: UIViewController {

    @IBOutlet weak var txtAdresse: UITextField!
    @IBOutlet weak var txtVille: UITextField!
    @IBOutlet weak var txtCP: UITextField!
    @IBOutlet weak var txtPays: UITextField!
    
    var coordonneesDestination : CLLocationCoordinate2D?
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    @IBAction func btnAfficherTrajet(_ sender: UIButton) {
        
        let adresse = "\(txtAdresse.text!),\(txtCP.text!), \(txtVille.text!), \(txtPays.text!)"
        
        let geocoder = CLGeocoder()
        
        geocoder.geocodeAddressString(adresse){
        (placemarks,erreur) in
            if erreur != nil{
                print("code erreur geocoder:\(erreur!.localizedDescription)")
            }
            else if placemarks != nil && placemarks!.count > 0
            {
                let placemark = placemarks![0]
                let location = placemark.location
                self.coordonneesDestination = location!.coordinate
                
                
                print("les coordonnées du premier point trouvé")
                print(" ")
                self.afficherMap()
            }
        }
        
    }
    
    func afficherMap()
    {
        //dictionnaire avec les clés spécifiques et avec les valeurs saisies
        let dicoAdresseDestination = [
            CNPostalAddressStreetKey: txtAdresse.text!,
            CNPostalAddressCityKey: txtVille.text!,
            CNPostalAddressPostalCodeKey: txtCP.text!,
            CNPostalAddressCountryKey: txtPays.text!]
        
        //réunir les coordonnées trouvées par le geocoder et les saisies dans un seul object
        let placemarkDestination = MKPlacemark(coordinate: coordonneesDestination!, addressDictionary: dicoAdresseDestination)
        
        //créer un objet de type MKMapItem
        let mkMapItem = MKMapItem(placemark: placemarkDestination)
        
        //PRéCISER DES éVENTUELLE OPTIONS DE PARCOUR
        // ces options doivent être données aussi dans un dictionnaire
        //avec des clés spécifiques
        let options = [MKLaunchOptionsDirectionsModeKey:MKLaunchOptionsDirectionsModeDriving]
        
        
        mkMapItem.openInMaps(launchOptions : options)
        
        
    }


}

